homeassistant/components/camera/local_file.py

```diff
@@ -61,7 +75,30 @@ def camera_image(self):
             _LOGGER.warning("Could not read camera %s image from file: %s",
                             self._name, self._file_path)
 
+    def check_file_path_access(self, file_path):
+        """Check that filepath given is readable."""
+        if not os.access(file_path, os.R_OK):
+            _LOGGER.warning("Could not read camera %s image from file: %s",
+                            self._name, file_path)
+
+    def update_file_path(self, file_path):
+        """Update the file_path."""
+        self.check_file_path_access(file_path)
+        self._file_path = file_path
+        self.schedule_update_ha_state()
+
     @property
     def name(self):
         """Return the name of this camera."""
         return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the camera state attributes."""
+        attr = super(LocalFile, self).device_state_attributes
```

Comment:
This is a lot of complicated things. You can just ignore super and just do:

```python
return {
  'file_path': self._file_path
}
```

Reply:
Sorry, I suggested to do this in a previous review comment (https://github.com/home-assistant/home-assistant/pull/13976#discussion_r183617856). I want apologize if that was wrong.

My train of thought was: the parent camera base class sets some attributes of which one seems important to support: [`auth_token`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/camera/__init__.py#L333). From what I understand, usually authentication is done with Home Assistant's HTTP Header. But I guess to support using the proxy for external integration, `auth_token` is shown in the attirbutes.